### DISCUSSION TITLE
Forgot password request not working behind NGINX proxy

### DISCUSSION THREAD

**Discussion by JamesHMcKay** [strapi/strapi#7333] [Legacy Topics]
Hello,

I have Strapi deployed using NGINX and pm2, and it works very well except for the forgot password request.

**Description**
POST request:
```
https://<strapi_url>/auth/forgot-password/
```
with body 
```
{"email": "<email>"}
```

- returns expected response if `<email>` is either not a valid email or not one of the registered users.  
- 504 Gateway time-out if email is that of a valid registered user.

So it appears the request is getting to Strapi, but it is unable to send out the password reset email if that is the necessary action, and eventually times out.  Therefore this is more of an issue with my NGINX set up (which follows the [documentation](https://strapi.io/documentation/v3.x/deployment/nginx-proxy.html#nginx-proxying)) than Strapi itself, so I am posting this here.

**Configuration**

My NGINX config is:

```
server {
 server_name <server_name>;

 location / {
        proxy_pass http://localhost:1337;
        proxy_http_version 1.1;
        proxy_set_header X-Forwarded-Host $host;
        proxy_set_header X-Forwarded-Server $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header Host $http_host;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_pass_request_headers on;
 }
    listen 443 ssl;
    ssl_certificate /etc/letsencrypt/live/...
    ssl_certificate_key /etc/letsencrypt/live/...
    include /etc/letsencrypt/options-ssl-nginx.conf;
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
}
```

I have also tested the password reset works when not behind NGINX by running the same build locally.

Has anyone else come across the same issue, or have currently got the Strapi password reset working when using NGINX and pm2?

- nginx version: 1.10.3 (Ubuntu)
- strapi version: 3.0.1
- pm2 version: 4.4.0

Any help or advice would be really appreciated, I have spent quite some time looking into this and can't seem to find a solution!

**Comment by derrickmehaffy** [comment]
How are you sending out the reset emails? (provider). For outbound requests, that doesn't involve Nginx. If the user requests their password to be reset -> proxied to Strapi which then triggers Strapi to send out the email via the provider (outbound request to send an email or request a provider service to send an email), unless your issue is with the user clicking the link in their email?

**Reply by JamesHMcKay** [reply]
Thanks, that makes a lot of sense.  I'll mark this as an answer since it clears up my question regarding the NGINX configuration.

In the end the issue was due to AWS blocking port 25 by default.  If anyone has the same issue the process to get the restriction removed is described here, https://aws.amazon.com/premiumsupport/knowledge-center/ec2-port-25-throttle/, and takes about two days.
